src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -2995,7 +2995,8 @@ public override void Visit(FirstNameNode node)
                 UpdateBindKindUseFlags(lookupInfo.Kind);
 
                 // Update statefulness of global datasources excluding dynamic datasources.
-                if (lookupInfo.Kind == BindKind.Data && !_txb._glue.IsDynamicDataSourceInfo(lookupInfo.Data))
+                if ((lookupInfo.Kind == BindKind.Data && !_txb._glue.IsDynamicDataSourceInfo(lookupInfo.Data)) ||
```

Comment:
>ookupInfo.Kind == BindKind.Dat [](http://example.com/codeflow?start=22&length=30)

May be use IsInfoKindDataSource

Reply:
It's for NameInfo but we need some helper like that.

Reply:
good call